tests/nz/net/ultraq/thymeleaf/tests/includes/Include-NestedFragmentInfiniteLoop.thtest

```diff
@@ -8,29 +8,26 @@
 %INPUT
 <html>
 <body>
-	<div layout:fragment="comment">
-		Comment structure
-		<div layout:fragment="comment-details">Default comment details</div>
-	</div>
-
-	<div layout:include="~{::comment}">
+	<div layout:include="~{comments :: comment}">
 		<div layout:fragment="comment-details">
 			Top level comment
-			<div layout:include="~{::comment}">
+			<div layout:include="~{comments :: comment}">
 				<div layout:fragment="comment-details">Nested comment</div>
 			</div>
 		</div>
 	</div>
 </body>
 </html>
 
+%INPUT[comments]
+	<div layout:fragment="comment">
+		Comment structure
+		<div layout:fragment="comment-details">Default comment details</div>
+	</div>
+
 %OUTPUT
 <html>
 <body>
-	<div>
-		Comment structure
-		<div>Default comment details</div>
-	</div>
```

Comment:
I'm a bit worried about the changes that were required to the "infinite loop" test files, as I think the change makes them different enough from the original bug that was raised: https://github.com/ultraq/thymeleaf-layout-dialect/issues/178  Do the changes made to `FragmentMap` alter the behaviour here?